pingcap/tidb
# expression: fix uuid generate duplicate value with multi-node

### What problem does this PR solve? 

In master `select uuid()` will got this result:

```
mysql> select uuid();
+--------------------------------------+
| uuid()                               |
+--------------------------------------+
| ad044d85-7dee-11e9-8000-000000000000 |
+--------------------------------------+
1 row in set (0.00 sec)
```
the question is `000000000000` part, it should node an IEEE 802 node number for RFC 4122 version-1

but has be `000000000000`, the root question is https://github.com/myesui/uuid/blob/master/generator.go#L21 init a generator but doesn't call https://github.com/myesui/uuid/blob/master/generator.go#L230 to init node value from https://github.com/myesui/uuid/blob/master/generator.go#L508

as result, mutiple node generate uuid maybe duplicate if they be called as same time.

### What is changed and how it works?

the easiest way to fix this question is call `NewGenerator` upper case method to init, instead of use `uuid.NewV1`, but after some look NewGenerator doesn't handle the question interface can not obtain, after thinking maybe we should change to https://github.com/google/uuid which 's impl is more like RFC's requirement https://github.com/google/uuid/blob/master/node.go#L39

ps: mysql's implement https://github.com/mysql/mysql-server/blob/6b10960e9d2e606b2774143da60903d641ac542f/sql/item_strfunc.cc#L3898 also will use random nodeid when interface can not  obtain too, https://dev.mysql.com/doc/refman/8.0/en/miscellaneous-functions.html#function_uuid, and uuid can be duplicate but it should be very low probability

at last, our ddl use `uuid.NewV4`  don't have this problem will keep use old lib.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test 

after this patch will got 

```
+--------------------------------------+
| uuid()                               |
+--------------------------------------+
| 08fbb5d0-7df1-11e9-80de-9cb6dx9073cb |
+--------------------------------------+
```

Code changes

- impl change
- add go.mod

Side effects

 - N/A

"Breaking backward compatibility" should not be problem, V1 use timestamp and new version will add real not it should not duplicate to old generated data by old tidb.

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pingcap/tidb/10590)
